compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -71,28 +83,13 @@ static SmallVector<Value> getInputOutputOperands(linalg::LinalgOp &linalgOp) {
 static FailureOr<SmallVector<Value>> getOperandsFromDefOp(
     linalg::LinalgOp &linalgOp) {
   SmallVector<Value> operands;
-  // For matmul only dispatch, we only want to fetch the input operand of the
-  // pack ops.
-  auto candidateOperands = isMatmulProducerOfElementwise(linalgOp)
-                               ? linalgOp->getOperands()
-                               : linalgOp.getDpsInputs();
-  for (auto operand : candidateOperands) {
-    auto defOp = operand.getDefiningOp();
-    if (!defOp) {
+  for (Value input : linalgOp.getDpsInputs()) {
+    auto defOp = input.getDefiningOp();
+    // The defining op has to be a pack op, fail otherwise.
+    if (!defOp || !isa<tensor::PackOp>(defOp)) {
```

Comment:
Just use `getDefiningOp<tensor::PackOp>()` - and in this case `auto` would be correct else we should've used `Operation*` if we only use `getDefiningOp()`.